bundles/org.openhab.binding.sonnen/src/main/resources/OH-INF/thing/thing-types.xml

```diff
@@ -0,0 +1,137 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<thing:thing-descriptions bindingId="sonnen"
+	xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+	xmlns:thing="https://openhab.org/schemas/thing-description/v1.0.0"
+	xsi:schemaLocation="https://openhab.org/schemas/thing-description/v1.0.0 https://openhab.org/schemas/thing-description-1.0.0.xsd">
+
+	<!-- Sonnen Thing Type -->
+	<thing-type id="sonnenbattery">
+		<label>sonnen Binding Thing</label>
+		<description>Thing for sonnen battery connection</description>
+
+		<channels>
+			<channel id="BatteryCharging" typeId="batteryCharging"/>
+			<channel id="BatteryDischarging" typeId="batteryDischarging"/>
+			<channel id="BatteryFeedIn" typeId="batteryFeedIn"/>
+			<channel id="BatteryDispense" typeId="batteryDispense"/>
+			<channel id="Consumption" typeId="consumption"/>
+			<channel id="GridFeedIn" typeId="gridFeedIn"/>
+			<channel id="GridReceive" typeId="gridReceive"/>
+			<channel id="SolarProduction" typeId="solarProduction"/>
+			<channel id="BatteryLevel" typeId="batteryLevel"/>
+			<channel id="FlowConsumptionBattery" typeId="flowConsumptionBattery"/>
+			<channel id="FlowConsumptionGrid" typeId="flowConsumptionGrid"/>
+			<channel id="FlowConsumptionProduction" typeId="flowConsumptionProduction"/>
+			<channel id="FlowGridBattery" typeId="flowGridBattery"/>
+			<channel id="FlowProductionBattery" typeId="flowProductionBattery"/>
+			<channel id="FlowProductionGrid" typeId="flowProductionGrid"/>
+		</channels>
+
+		<config-description>
+			<parameter name="hostIP" type="text" required="true">
+				<context>network-address</context>
+				<label>IP Address</label>
+				<description>Please add the IP Address of your sonnen battery</description>
+			</parameter>
+			<parameter name="refreshInterval" type="integer" unit="s" min="0" max="1000">
+				<label>Refresh Interval</label>
+				<description>How often in seconds the sonnen battery should schedule a refresh after a channel is linked to an item.
+					Valid input is 0 - 1000</description>
+				<advanced>true</advanced>
+				<default>30</default>
+			</parameter>
+		</config-description>
+	</thing-type>
+
+	<!-- sonnen Channel Type -->
+	<channel-type id="batteryCharging">
+		<item-type>Switch</item-type>
+		<label>Battery Charging</label>
+		<description>Indicates if the Battery is charging at that moment</description>
+		<state readOnly="true"/>
+	</channel-type>
+	<channel-type id="batteryDischarging">
+		<item-type>Switch</item-type>
+		<label>Battery Discharging</label>
+		<description>Indicates if the Battery is discharging at that moment</description>
+		<state readOnly="true"/>
+	</channel-type>
+	<channel-type id="batteryFeedIn">
+		<item-type>Number:Energy</item-type>
+		<label>Battery Feed In</label>
+		<description>Indicates the actual current feeding to the Battery in watt. Otherwise 0</description>
+		<state readOnly="true"/>
```

Comment:
Generally, it is a good practice to propose a default pattern for any number channel. It allows to you to give an indication to the precision. Like for example:
`<state readOnly="true" pattern="%.2f %unit%"/>`
I let you adjust how many decimals you think is appropriate.
By using `%unit%`, the default unit for the user will be the unit you used to set the channel state which is in watts for all your channels I believe.
Please do it for all your number channels